src/main/java/com/simondata/pouroversql/ExtractorFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019-present, Simon Data, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.simondata.pouroversql;
+
+import com.simondata.pouroversql.clients.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ExtractorFactory
+ *
+ * Factory class to build the correct type of Extractor based on input.
+ */
+public class ExtractorFactory {
+    private static final Logger logger = LoggerFactory.getLogger(ExtractorFactory.class);
+
+    public static AbstractExtractor makeExtractor(ParamsHolder paramsHolder) {
+        ExtractorEngine extractorEngine = ExtractorEngine.byName(paramsHolder.getExtractorType());
+        paramsHolder.setExtractorEngine(extractorEngine);
+        return ExtractorFactory.makeExtractor(extractorEngine, paramsHolder);
```

Comment:
I believe you can use `this.makeExtractor...` in Java.

Reply:
Tried updating to go that route, but with how we're calling it (being consistent with ClientFactory) and it being a static method, it can't use "this", has to be explicit with using ExtractorFactory